src/protonpack/java/com/codepoetics/protonpack/AggregatingSpliterator.java

```diff
@@ -15,8 +17,8 @@
     private List<I> currentSlide = new ArrayList<>();
 
     AggregatingSpliterator(Spliterator<I> source, BiPredicate<List<I>, I> predicate) {
-        this.source = source;
-        this.condition = predicate;
+        this.source = requireNonNull(source);
+        this.condition = requireNonNull(predicate);
```

Comment:
There's still the windowed spliterator. If you give some dumb parameters, such as a negative window size, it hangs in an infinite loop. I don't think it's a desirable behavior.